verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -138,15 +140,50 @@ void IndexingFactsTreeExtractor::ExtractModule(
     Visit(module_item_list);
   }
 
-  facts_tree_context_.back()->NewChild(module_node);
+  facts_tree_context_.top().NewChild(module_node);
 }
 
 void IndexingFactsTreeExtractor::ExtractModuleHeader(
     const verible::SyntaxTreeNode& node) {
   const verible::TokenInfo& module_name_token = GetModuleNameToken(node);
   const Anchor module_name_anchor(module_name_token, context_.base);
 
-  facts_tree_context_.back()->Value().AppendAnchor(module_name_anchor);
+  facts_tree_context_.top().Value().AppendAnchor(module_name_anchor);
+
+  const verible::SyntaxTreeNode* port_list = GetModulePortDeclarationList(node);
+
+  if (port_list != nullptr) {
+    std::vector<verible::TreeSearchMatch> port_names =
```

Comment:
where possible declare these objects as `const`.  Helps readers know it won't be modified below.